docs/plugins/inputs/snmp.asciidoc

```diff
@@ -64,6 +64,37 @@ $ smidump --level=1 -k -f python RFC1213-MIB > RFC1213-MIB.dic
 
 Note that the resulting file as output by `smidump` must have the `.dic` extension.
 
+As `smidump` only looks for mib dependencies in its pre-configured path lists as defined in the documentaion in the https://www.ibr.cs.tu-bs.de/projects/libsmi/smi_config.html["MODULES LOCATIONS"] section, you may need to provide the paths to locations of MIBs in your particular environment to avoid a `failed to locate MIB module` error.
+
+Two ways to achieve this are by using an environment variable or a config file to provide the additional path configuration.
+
+Set the `SMIPATH` env var with the path to your mibs ensuring you include a prepended colon `:` for the path, for example:
+
+[source,sh]
+-----
+$ SMIPATH=":/path/to/mibs/" smidump -k -f python CISCO-PROCESS-MIB.mib > CISCO-PROCESS-MIB_my.dic
+-----
+
+NOTE: SMI docs on the prepended colon:
+
+  The ‘path’ command argument and the environment variable either start with a path separator character (‘:’ on UNIX-like systems, ‘;’ on MS-Windows systems) to append to the path, or end with a path separator character to prepend to the path, or otherwise completely replace the path.
+
+
+The other way is to create a configuration file eg `smi.conf` with the `path` option.
+
+[source,sh]
+-----
+path :/path/to/mibs/
+-----
+
+and use that config with smidump:
+
+[source,sh]
+-----
+$ smidump -c smi.conf -k -f python CISCO-PROCESS-MIB.mib > CISCO-PROCESS-MIB_my.dic
+-----
+
```

Comment:
```suggestion
[id="plugins-{type}s-{plugin}-locate-mibs"]
===== Preventing a `failed to locate MIB module` error 
3 weeks ago

[[DOCS] add troubleshooting help for failed to locate MIB module smidu…](https://github.com/karenzone/logstash-input-snmp/commit/244ab1d2784a1f279ec0c47352042c360a09ea30)

2 days ago

[Wording tweak](https://github.com/karenzone/logstash-input-snmp/commit/f8c2c2dace9b3c3bbe95162e439cdc73135a2b30)
The `smidump` function looks for mib dependencies in its pre-configured path lists.
You may need to provide the paths to locations of MIBs in your particular environment.
3 weeks ago

[[DOCS] add troubleshooting help for failed to locate MIB module smidu…](https://github.com/karenzone/logstash-input-snmp/commit/244ab1d2784a1f279ec0c47352042c360a09ea30)

2 days ago

[Wording tweak](https://github.com/karenzone/logstash-input-snmp/commit/f8c2c2dace9b3c3bbe95162e439cdc73135a2b30)
You can avoid  to avoid a `failed to locate MIB module` error by providing the additional path configuration information with:
last week

[Doc: Clarify troubleshooting for missing MIB](https://github.com/karenzone/logstash-input-snmp/commit/74c09ef5aafe4e81d33641f245896282e930e420)

* an environment variable, or 
* a config file to provide the additional path configuration.

2 days ago

[Wording tweak](https://github.com/karenzone/logstash-input-snmp/commit/f8c2c2dace9b3c3bbe95162e439cdc73135a2b30)
See the "MODULE LOCATIONS" section of the https://www.ibr.cs.tu-bs.de/projects/libsmi/smi_config.html[smi_config documentation] for more info. 

last week

[Doc: Clarify troubleshooting for missing MIB](https://github.com/karenzone/logstash-input-snmp/commit/74c09ef5aafe4e81d33641f245896282e930e420)
[id="plugins-{type}s-{plugin}-env-var"]
===== Option 1: Use an environment variable 

Set the `SMIPATH` env var with the path to your mibs.
Be sure to include a prepended colon (`:`) for the path.
3 weeks ago

[[DOCS] add troubleshooting help for failed to locate MIB module smidu…](https://github.com/karenzone/logstash-input-snmp/commit/244ab1d2784a1f279ec0c47352042c360a09ea30)

[source,sh]
-----
last week

[Doc: Clarify troubleshooting for missing MIB](https://github.com/karenzone/logstash-input-snmp/commit/74c09ef5aafe4e81d33641f245896282e930e420)
$ SMIPATH=":/path/to/mibs/" smidump -k -f python CISCO-PROCESS-MIB.mib > CISCO-PROCESS-MIB_my.dic <1>
3 weeks ago

[[DOCS] add troubleshooting help for failed to locate MIB module smidu…](https://github.com/karenzone/logstash-input-snmp/commit/244ab1d2784a1f279ec0c47352042c360a09ea30)
-----
last week

[Doc: Clarify troubleshooting for missing MIB](https://github.com/karenzone/logstash-input-snmp/commit/74c09ef5aafe4e81d33641f245896282e930e420)
<1> Notice the colon that preceeds the path definition. 
3 weeks ago

[[DOCS] add troubleshooting help for failed to locate MIB module smidu…](https://github.com/karenzone/logstash-input-snmp/commit/244ab1d2784a1f279ec0c47352042c360a09ea30)

last week

[Doc: Clarify troubleshooting for missing MIB](https://github.com/karenzone/logstash-input-snmp/commit/74c09ef5aafe4e81d33641f245896282e930e420)
[id="plugins-{type}s-{plugin}-mib-config"]
===== Option 2: Provide a coniguration file
3 weeks ago

[[DOCS] add troubleshooting help for failed to locate MIB module smidu…](https://github.com/karenzone/logstash-input-snmp/commit/244ab1d2784a1f279ec0c47352042c360a09ea30)

last week

[Doc: Clarify troubleshooting for missing MIB](https://github.com/karenzone/logstash-input-snmp/commit/74c09ef5aafe4e81d33641f245896282e930e420)
The other approach is to create a configuration file with the `path` option. For example, you could create a file called `smi.conf`.
3 weeks ago

[[DOCS] add troubleshooting help for failed to locate MIB module smidu…](https://github.com/karenzone/logstash-input-snmp/commit/244ab1d2784a1f279ec0c47352042c360a09ea30)

[source,sh]
-----
path :/path/to/mibs/
-----

last week

[Doc: Clarify troubleshooting for missing MIB](https://github.com/karenzone/logstash-input-snmp/commit/74c09ef5aafe4e81d33641f245896282e930e420)
Use the config with smidump:
3 weeks ago

[[DOCS] add troubleshooting help for failed to locate MIB module smidu…](https://github.com/karenzone/logstash-input-snmp/commit/244ab1d2784a1f279ec0c47352042c360a09ea30)

[source,sh]
-----
$ smidump -c smi.conf -k -f python CISCO-PROCESS-MIB.mib > CISCO-PROCESS-MIB_my.dic
-----
```